libraries/AP_HAL_PX4/CAN.cpp

```diff
@@ -0,0 +1,1035 @@
+/*
+ * Copyright (C) 2014 Pavel Kirienko <pavel.kirienko@gmail.com>
+ *
+ * With modifications for Ardupilot CAN driver
+ * Copyright (C) 2017 Eugene Shamaev
+ */
+
+//#define CONFIG_HAL_BOARD HAL_BOARD_PX4
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/system.h>
+
+#include <AP_BoardConfig/AP_BoardConfig.h>
+
+#include <cassert>
+#include <cstring>
+#include "CAN.h"
+
+#include <nuttx/irq.h>
+#include <unistd.h>
+#include <drivers/drv_hrt.h>
+
+#include <assert.h>
+
+#include "Scheduler.h"
+
+/*
+ * FOR INVESTIGATION:
+ * AP_HAL::micros64() was called for monotonic time counter
+ * pavel-kirienko: This will work as long as we don't need to synchronize the autopilot's own clock with an external
+ * time base, e.g. a GNSS time provided by an external GNSS receiver. Libuavcan's STM32 driver supports automatic time
+ * synchronization only if it has a dedicated hardware timer to work with.
+ */
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_PX4
+
+#define STM32_SYSCLK_FREQUENCY  168000000ul
+#define STM32_HCLK_FREQUENCY    STM32_SYSCLK_FREQUENCY
+#define STM32_PCLK1_FREQUENCY   (STM32_HCLK_FREQUENCY/4)
+
+#ifndef UAVCAN_NULLPTR
+# if UAVCAN_CPP_VERSION >= UAVCAN_CPP11
+#  define UAVCAN_NULLPTR nullptr
+# else
+#  define UAVCAN_NULLPTR NULL
+# endif
+#endif
+
+#include "CAN.h"
```

Comment:
Actually no. You can't include CAN drivers without UAVCAN precisely because they are based in libuavcan... So if HAL_WITH_UAVCAN isn't defined you can't include this file or the build will break.